File: lab1/Models/Project.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TRS.Models
{
    public class Project
    {
        public string code { get; set; }
        public string manager { get; set; }
        [Required(ErrorMessage = "Please enter name")]
        public string name { get; set; }
        public int budget { get; set; }
        public bool active { get; set; } = true;
        public List<string> subactivities { get; set; } = new List<string>();

        public void AddSubactivity(string subactivity)
        {
            subactivities.Add(subactivity);
        }

        public void DeleteSubactivity(string subactivity)
        {
            subactivities.Remove(subactivity);
        }
    }
}

File: lab4/trs/src/Project.js
import React, { Fragment } from 'react';

function Project(props) {
    let project = props.project;
    return (
        <Fragment>
            <tr>
                <td>{project.code}</td>
                <td>{project.manager}</td>
                <td>{project.name}</td>
                <td>{project.budget}</td>
                <td>{project.active ? <p>Active</p> : <p>Inactive</p>}</td>
                <td>{project.subactivities.map((s) => <p>{s}</p>)}</td>
                {/* <td>project.code</td> */}
            </tr>
        </Fragment>
    );
};

export default Project;